firewood/src/merkle.rs

```diff
@@ -369,18 +410,18 @@ impl<T: HashedNodeReader> Merkle<T> {
                     };
 
                     let inserted = seen.insert(child_addr);
-                    if !inserted {
+                    if inserted {
```

Comment:
nit: not sure I agree with this change, as I'd prefer to handle errors closer to where the problem occurred, but I'm not completely against it